pydy/system.py

```diff
@@ -340,22 +341,25 @@ def times(self):
         equations of motion are integrated, numerically.
 
         The object should be in a format which the integration module to be
-        used can accept. Since this attribute is not checked for
-        compatibility, the user becomes responsible to supply it correctly.
+        used can accept.
         """
         return self._times
 
     @times.setter
     def times(self, new_times):
-        self._times = new_times
+        self._times = np.asarray(new_times)
+        self._check_times(self._times)
 
     def _check_times(self, times):
-        """
-        Very basic checking.
-        TODO: add more checking
-        """
-        if len(times) == 0:
+        if len(times.shape) == 0:
```

Comment:
@moorepants 
```python
>>> x = 4
>>> x = np.asarray(x)
>>> x
array(4)
>>> x.shape
()
>>> len(x.shape)
0
```
```asarray``` does not throw an error for a non array like input. I think it gives an array of zero dimension. Hence checking dimension length gives 0. Please correct me if I am wrong,